src/main/java/org/folio/circulation/resources/CheckInProcessAdapter.java

```diff
@@ -102,4 +116,23 @@
 
     return completedFuture(succeeded(item));
   }
+
+  CompletableFuture<HttpResult<CheckInProcessRecords>> sendCheckInPatronNotice(CheckInProcessRecords records) {
+    if (records.getLoan() == null) {
+      return CompletableFuture.completedFuture(HttpResult.succeeded(records));
+    }
+    return patronNoticePolicyRepository.lookupNoticePolicy(records.getLoan())
+      .thenApply(r -> r.next(policy -> {
+        sendCheckOutPatronNoticeWhenPolicyFound(records, policy);
+        return HttpResult.succeeded(records);
+      }));
+  }
+
+  private void sendCheckOutPatronNoticeWhenPolicyFound(CheckInProcessRecords records, PatronNoticePolicy patronNoticePolicy) {
+    Loan loan = records.getLoan();
+    List<NoticeDescriptor> noticeDescriptors =
+      patronNoticePolicy.lookupLoanNoticeDescriptor(NoticeEventType.CHECK_IN, NoticeTiming.UPON_AT);
+    JsonObject noticeContext = patronNoticeService.createNoticeContextFromLoan(loan);
+    patronNoticeService.sendPatronNotice(noticeDescriptors, loan.getUserId(), noticeContext);
```

Comment:
Can there be multiple notices (in the same policy) for a check in?

Reply:
Yes, the schema for patron notice policy supports it.

Reply:
Fair enough, that seems counter intuitive to me, maybe that is a follow up question for Darcy around notice policies?

Reply:
I think so, now there is no validation for that.